### PR TITLE
Annotate Day 121

### PR DESCRIPTION
**Pull request by miblodelcarpio**
There is still more work to do, but the majority is in

### CODE REVIEW

File: src/documents/videos/game-architecture/day121.html.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+title: "Rendering in Tiles (Marathon)"
+videoId: "kZlPYka1T0g"
+markers:
+    "1:36": "Reintroducing the Intel Architecture Code Analyzer"
+    "10:46": "A long time ago, in RAD's source tree"
+    "13:00": "blowtard"
```
Comment (dspecht) [inline src/documents/videos/game-architecture/day121.html.md]: maybe say its a xbox360 cpu analyser

File: src/documents/videos/game-architecture/day121.html.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    "30:56": "Looking at how to get our cycle count down"
+    "32:21": "Manually unroll the Fetch / Sample loop"
+    "36:30": "Group by Sample"
+    "37:27": "Use _mm_setr_ps as suggested by Fabian a long time ago"
+    "41:47": "Five minute warning!"
+    "42:14": "Taking a look at the total throughput count"
+    "43:18": "Casey needs some more soya [sic] milk"
+    "44:17": "Could we do a load once, and grab out the two values that we needed?"
+    "45:48": "Day 121 Blackboard: Explanation of possible texel loading optimisation"
+    "46:47": "Q&A begins now"
```
Comment (dspecht) [inline src/documents/videos/game-architecture/day121.html.md]: doesn't actually begin here, so no using the popups if its not actually the correct thing.

File: src/documents/videos/game-architecture/day121.html.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    "50:32": "Figuring out how the compiler is loading the texel data"
+    "1:00:18": "This is fine, then"
+    "1:01:01": "We multiply by TexturePitch and sizeof(uint32) four-wide manually, which is stupid"
+    "1:01:06": "Shift up FetchX_4x by 2, rather than multiply by sizeof(uint32)"
+    "1:03:40": "Premultiply FetchY_4x by TexturePitch_4x"
+    "1:04:07": "Give the compiler the wide stuff so that it can see it as wide"
+    "1:11:21": "_mm_mul_epi32 does not do integer × integer"
+    "1:13:43": "Port pressure (we're back to InterIteration)"
+    "1:16:47": "Stream has ended"
```
Comment (dspecht) [inline src/documents/videos/game-architecture/day121.html.md]: again it still goes on so this is pointless and not useful

File: src/documents/videos/game-architecture/day121.html.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    "1:40:47": "Implementation of interleaved scanlines, in readiness for hyperthreading"
+    "1:46:36": "Explaining the logic of interleaved scanlines"
+    "1:52:37": "Updating compiler directives for folks who use LLVM"
+    "1:55:20": "Implementation of frame buffer divisions, in readiness for multi-core processing"
+    "2:05:30": "Go to Disassembly of DrawRectangleSlowly in order to diagnose bogus cycle count"
+    "2:10:04": "Frame buffer divisions, continued"
+    "2:20:50": "Introduce GetClampedRectArea"
+    "2:22:12": "Problematic thing: "Our convention for rectangles before was that they did not include their final value""
+    "2:27:33": "Fix the cycle counter for DrawRectangleSlowly again"
+
```
Comment (dspecht) [inline src/documents/videos/game-architecture/day121.html.md]: no break should be here so need to remove this extra line.

File: src/documents/videos/game-architecture/day121.html.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    "2:30:56": "Introduce ExpandRectangleToIncludePoint"
+    "2:30:56": "TODO(matt): WHAT IS HAPPENING HERE?!"
+    Take each individual point of the 4 corners of our potentially rotated quad
+    that we're texture mapping, and it's trying to make sure that it gets the min /
+    max bounds of those
+
```
Comment (dspecht) [inline src/documents/videos/game-architecture/day121.html.md]: again a extra line, and we need to not do this mutli line thing, though need to view that for a idea if non is gathered.

File: src/documents/videos/game-architecture/day121.html.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    "3:02:03": "Dancing trees"
+    "3:03:03": "Change our loads and stores to no longer be unaligned"
+    "3:04:05": "Assess performance difference and revert back to the unaligned load and store instructions"
+    "3:05:12": "Make sure that we actually always fill the real clip region and not write outside the clip region"
+    "3:07:10": "Explanation of our options for filling the pixels"
+    "3:09:12": "Implementation of alignment to the ending edge"
+    "3:16:48": "Clip the leading edge"
+    "3:19:41": "Explanation of ClipMask"
+    "3:21:33": "Try setting StartupClipMask by using _mm_srli_si128"
+    "3:22:28": "// TODO(casey): This is stupid."
```
Comment (dspecht) [inline src/documents/videos/game-architecture/day121.html.md]: 63 and 54 probably need to be looked at how useful they are.

File: src/documents/videos/game-architecture/day121.html.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    "3:26:10": "Early-out the FillRect tests"
+    "3:30:01": "Start passing ClipRect through to DrawRectangleQuickly"
+    "3:35:35": "Moment of realisation, with introduction of the InvertedInfinityRectangle"
+    "3:37:48": "Temporarily adjust ClipRect in order to avoid a crash"
+    "3:39:24": "Introduce TiledRenderGroupToOutput outside of the timer"
+    "3:43:57": "Update DrawRectangle to take the clipping information"
+    "3:47:18": "Update DrawRectangle{,Quickly} to use the Even / Odd information"
+    "3:49:20": "Break the screen up into pieces and render them separately"
+    "3:54:34": "Stretch your legs, Casey"
```
Comment (dspecht) [inline src/documents/videos/game-architecture/day121.html.md]: 72, maybe useful but more joking.

File: src/documents/videos/game-architecture/day121.html.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    "4:04:57": "mmozeiko Q: Will you revert yesterday changes where you changed bilinear pixel unpacking code from float mul to int mul? It was faster with float mul."
+    "4:05:24": "an0nymal Q: How many more marathon streams will we have? I thoroughly enjoyed the 4+ hours today."
+    "4:05:47": "quikligames Q: You should give a big thanks to @Rygorous for sticking around and trying to give you tips knowing full well that you wouldn't see them in chat"
+    "4:06:07": "mmozeiko Q: would it be better to have tile sizes always divisible by 4 horizontally (or even 16 to be cache aligned), then there will be no need to deal with alignment and masking?"
+    "4:07:07": "rygorous Q: (clip) one too few pixels. look at the edge of the screen."
+    "4:09:20": "rygorous Q: just pretty sure I saw glitchiness/off-by-1-pixel stuff near the edges but it might've been the video encoding"
+    "4:11:08": "mmozeiko Q: (tile size %4) - not masking for textures, but ClipMask variable"
+    "4:13:30": "@abnercoimbre Q: holy crap. our 1st marathon."
+    "4:13:50": "Time for Casey to go to bed"
+---
```
Comment (dspecht) [inline src/documents/videos/game-architecture/day121.html.md]: maybe clean up this end for not useful stamps,

**Comment by miblodelcarpio** [comment]
Please don't merge this, collaborators. There's still work to do.

File: src/documents/videos/game-architecture/day121.html.md (hunk 5 above)
Comment (miblodelcarpio) [inline src/documents/videos/game-architecture/day121.html.md]: This is the unfinished bit referred to in the commit message and which I mentioned in IRC. Fixed in a subsequent commit.

**Comment by miblodelcarpio** [comment]
This is now complete enough as far as I'm concerned, so ready to be checked and potentially merged.

**Pull request merged**